model/train_grnet.py

```diff
@@ -12,10 +12,24 @@
 
 torch.autograd.set_detect_anomaly(True)
 
+
+class log_space_L1_loss(torch.nn.Module):
+    def __init__(self, weight=None, size_average=True):
+        super(log_space_L1_loss, self).__init__()
+
+    def forward(self, reconstruction, target):
+        target_log = torch.log(torch.abs(target) + 1)
+
+        loss = torch.abs(torch.abs(reconstruction) - target_log)
+
+        sign_mask = torch.sign(reconstruction) * torch.sign(target) < 0
+        loss = torch.where(sign_mask, 2 * loss, loss)
```

Comment:
Why are we making 2*loss for the places where both locations (target and reconstruction) are inside and just loss for the rest (when they have different signs)?

Reply:
Yes, you are correct, the mask trick will make it meaningless, so maybe only keep line23?